### PR TITLE
App header

### PR DESCRIPTION
**Pull request by andisima**
# Description
What?
 - We've added the logout button with functionality.
 - We've added the Profile button, with no functionality.
 - We've added a top bar to add the mentioned buttons.

 - The logout button works to logout from the user's account.
- The profile button will work in the future to go to the user's profile.
 
 # Testing
 - Clone Repository
- Run Bundle Install
- Create a database
- Do a Migration
- Run the server
- Go to localhost:3000
- login or create an account
- Then you will be in the dashboard, in the upper left you'll see the created buttons.

### CODE REVIEW

**Review by jaimelr — COMMENTED** [review]
The header is a site component that is mostly expected to be available across the whole site. With the current approach, it is only going to be visible in the `index` page AKA in only one route of the whole application.

There are situations where you need different headers depending on the page the user is viewing, but still if the component is expected to be available in more than one page, I suggest you to use a partial that is rendered in the main `application.html.erb` layout.

For example:
```
<body>
  <% render 'header' %>
  <%= yield %>
</body>
```

The above code will render the header above all the content in your app, no matter the route or page the user is viewing (assuming we have a partial called `_header.html.erb` somewhere in the `app/views` folder.

Please let me know what you think of this approach and specially ping me if you have questions about how to implement it.

**Comment by jaimelr** [comment]
I would also suggest that for visual changes (user-facing) you add a before (if applies) and after screenshots of the page you are modifying. This would help the reviewer to get quick context on what are you changing in this PR.

File: spec/login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
```
Comment (jclopezdev) [inline spec/login_spec.rb]: I'd add this spec file within a "features" folder, what do you think?

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -1 +1,20 @@
-<%= link_to("log out", destroy_admin_session_path, method: :delete) %>
\ No newline at end of file
+<nav class="navbar navbar-dark navbar-expand-lg bg-dark flex-md-nowrap p-0 shadow">
+  <ul class="navbar-nav px-3">
+    <li class="nav-item text-nowrap">
+      <%= link_to("log out", destroy_admin_session_path, method: :delete , class:'nav-link') %>
```
[the remark this answers is not in the document]
Comment (andisima) [reply app/views/home/index.html.erb:4]: Modified.

File: app/views/home/index.html.erb [hunk 3]
```diff
@@ -1 +1,20 @@
...
+    </li>
+    <li class="nav-item text-nowrap">
+    <a class="nav-link text-wrap" href="#">Perfil</i></a>
```
[the remark this answers is not in the document]
Comment (andisima) [reply app/views/home/index.html.erb:7]: Modified, thanks!

File: app/views/home/index.html.erb [hunk 4]
```diff
@@ -1 +1,20 @@
...
+  <a class="navbar-brand col-sm-3 col-md-2 ml-auto" href="#">Control Almacen</a>
+</nav>
+
+<% flash.each do |type, msg| %>
+  <div class="<%= flash_class(type) %>">
+    <%= msg %>
+  </div>
+<% end %>
+
+ <!--<%= link_to("log out", destroy_admin_session_path, method: :delete) %>-->
```
[the remark this answers is not in the document]
Comment (andisima) [reply app/views/home/index.html.erb:19]: Modified, thanks!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -10,11 +10,6 @@
   </head>
 
   <body>
-    <% flash.each do |type, msg| %>
-      <div class="<%= flash_class(type) %>">
-        <%= msg %>
-      </div>
-    <% end %>
```
[the remark this answers is not in the document]
Comment (andisima) [reply app/views/layouts/application.html.erb]: We moved it again to the main layout page, thanks!

File: app/views/home/index.html.erb [hunk 6]
```diff
@@ -1 +1,20 @@
-<%= link_to("log out", destroy_admin_session_path, method: :delete) %>
\ No newline at end of file
+<nav class="navbar navbar-dark navbar-expand-lg bg-dark flex-md-nowrap p-0 shadow">
```
Comment (CarlosSe) [inline app/views/home/index.html.erb:1]: Hi Andrea you use the class "navbar-expand-lg" for that reason I think is necessary a button that collapses the content behind to hide or show it.
```suggestion
<nav class="navbar navbar-dark navbar-expand-lg bg-dark flex-md-nowrap p-0 shadow">
<button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarTogglerDemo01" aria-controls="navbarTogglerDemo01" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
```
Or for a nav always expanded
```suggestion
<nav class="navbar navbar-dark navbar-expand bg-dark flex-md-nowrap p-0 shadow">
```

File: Rakefile [hunk 7]
```diff
@@ -2,6 +2,10 @@
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
 require_relative 'config/application'
+require 'rubocop'
+require 'rubocop/rake_task'
+# require 'rubocop-rake'
```
Comment (pacheco-andres) [inline Rakefile:7]: If we are not using this line of code we must remove it

File: app/assets/stylesheets/application.css [hunk 8]
```diff
@@ -14,3 +14,6 @@
  *= require_tree .
  *= require_self
  */
+ .notice{
+     color:green;
```
Comment (pacheco-andres) [inline app/assets/stylesheets/application.css:18]: Check indentation and space after colon

**Review by pacheco-andres — CHANGES_REQUESTED** [review]
Review comments

File: app/views/home/index.html.erb (hunk 6 above)
Comment (alx1313) [reply app/views/home/index.html.erb:1]: I think its better to use nav always expanded instead of collapse it, mainly because we will need a sidebar, and in my opinion, sidebar would be better if it can collapse.
WDYT?

File: Rakefile (hunk 7 above)
Comment (alx1313) [reply Rakefile:7]: Agreed :smile:

File: app/views/home/index.html.erb (hunk 6 above)
Comment (CarlosSe) [reply app/views/home/index.html.erb:1]: I agree with you, just that usually the button to collapse sidebar is on navbar, the code doesn't define a navbar always expanded, define a navbar expanded on large screens "navbar-expand-lg"

File: app/assets/stylesheets/application.css (hunk 8 above)
Comment (alx1313) [reply app/assets/stylesheets/application.css]: solved!

**Comment by cfcortes** [comment]
@jaimelr @pacheco-andres al parecer ya se atendieron las observaciones. Nos ayudan con otra revisión y en su caso aprobación, por favor?

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -11,10 +11,8 @@
...
   <body>
     <% flash.each do |type, msg| %>
-      <div class="<%= flash_class(type) %>">
-        <%= msg %>
-      </div>
-    <% end %>
-    <%= yield %>
+  <div class="<%= flash_class(type) %>">
+    <%= msg %>
+  </div>
```
Comment (EduardoGHdez) [inline app/views/layouts/application.html.erb:16]: Can you pls fix the indentation here?

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -11,10 +11,8 @@
 
   <body>
     <% flash.each do |type, msg| %>
-      <div class="<%= flash_class(type) %>">
-        <%= msg %>
-      </div>
-    <% end %>
-    <%= yield %>
```
Comment (EduardoGHdez) [inline app/views/layouts/application.html.erb:18]: Any special reason for removing `yield` from here?

Comment (EduardoGHdez) [reply app/views/layouts/application.html.erb:18]: Without this, rails is not able to render the other views in the application layout

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -11,10 +11,8 @@
 
   <body>
     <% flash.each do |type, msg| %>
-      <div class="<%= flash_class(type) %>">
-        <%= msg %>
-      </div>
-    <% end %>
```
Comment (EduardoGHdez) [inline app/views/layouts/application.html.erb:17]: Without this, the block would not know where it finish

File: app/views/home/index.html.erb [hunk 12]
```diff
@@ -1 +1,17 @@
-<%= link_to("log out", destroy_admin_session_path, method: :delete) %>
+<nav class="navbar navbar-dark navbar-expand-lg bg-dark flex-md-nowrap p-0 shadow">
```
Comment (EduardoGHdez) [inline app/views/home/index.html.erb:1]: WDYT about moving this to a partial view and render it form the `application layout`?

This way we'd have the navbar in all view that uses this layout and in case we create another one, it'd be easy to use it too.

File: app/views/layouts/application.html.erb (hunk 9 above)
Comment (alx1313) [reply app/views/layouts/application.html.erb:16]: sure!

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,7 +1,21 @@
...
+        <li class="nav-item text-nowrap">
+          <%= link_to("log out", destroy_admin_session_path, method: :delete , class:'nav-link') %>
+        </li>
+        <li class="nav-item text-nowrap">
+        <a class="nav-link text-wrap" href="#">Profile</i></a>
+        </li>
+      </ul>
+      <a class="navbar-brand col-sm-3 col-md-2 ml-auto" href="#">Warehouse Control</a>
+    </nav>
+>>>>>>> Stashed changes
```
Comment (Ilyeo) [inline app/views/layouts/application.html.erb:17]: solve these conflicts please!

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,7 +1,19 @@
...
+      <ul class="navbar-nav px-3">
+        <li class="nav-item text-nowrap">
+          <%= link_to("log out", destroy_admin_session_path, method: :delete , class:'nav-link') %>
+        </li>
+        <li class="nav-item text-nowrap">
+        <a class="nav-link text-wrap" href="#">Profile</i></a>
+        </li>
+      </ul>
+      <a class="navbar-brand col-sm-3 col-md-2 ml-auto" href="#">Warehouse Control</a>
+    </nav>
```
Comment (Ilyeo) [inline app/views/layouts/application.html.erb:15]: this is curious, I mean inside the `<head>` tag. I am used to seeing the `nav` element into the `body` or into `header`, not into the `head`

Comment (dannharo) [reply app/views/layouts/application.html.erb:15]: yeah thats right, semantically this is not correct, this should be placed in the `body` tag

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -1,7 +1,19 @@
...
   <head>
-    <title>Control Almacen</title>
+
+    <nav class="navbar navbar-dark navbar-expand bg-dark flex-md-nowrap p-0 shadow">
+      <ul class="navbar-nav px-3">
+        <li class="nav-item text-nowrap">
+          <%= link_to("log out", destroy_admin_session_path, method: :delete , class:'nav-link') %>
+        </li>
+        <li class="nav-item text-nowrap">
+        <a class="nav-link text-wrap" href="#">Profile</i></a>
```
Comment (dannharo) [inline app/views/layouts/application.html.erb:11]: check this indentation, the `a` is inside the `li` tag

**Review by vzqzac — CHANGES_REQUESTED** [review]
Please edit your commits so that you have atomic semantic ones, for example:
Commits https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/7486c08b5f6bd989c8f367ff54d58694a0d0de06 https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/dfa4d8dfeeb7b4df5d25baa3944e50016cf12ad5 https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/2b36ee71ee1d56bc807cf92f5af99de4a1492eb6 and https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/70a2008e9a9b49e15b85e9775f5cfdbb52e9e5d3 should not exist by themselves, when fixing something that is living in a pull request (that is, it hasn't been merged yet) don't create new commits, instead, create fixup commits and then rebase the branch, [refer to this lecture for an example](https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html).

There are also some merge commits: https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/7e8dd256478654b7bd91d90b843d85925113ae3d and https://github.com/BrightCoders-Proyectos/control-almacen/pull/17/commits/bb4ae81dd5b3658530805c43d256ff25981e67a4. Again, this commits should not exist in a pull request, since they're not part of the actual proposed implementation, this might be due to a conflict when rebasing or using merge instead of rebase, please try to use rebase instead of merging to prevent this, and resolve conflicts as they show instead of merging to get rid of them.

If you have any troubles while working on these fixes please ask me or any other mentor for help, we're here to guide you if things become difficult when working on the project.

File: team-commit.txt [hunk 16]
```diff
@@ -15,4 +15,6 @@ Co-authored-by: CarlosSe <carlos.seggch@gmail.com>
 Co-authored-by: jorge27 <ja.garciavega@outlook.com>
 Co-authored-by: DiegoFigueroa98 <16460315@colima.tecnm.mx>
 Co-authored-by: pacheco-andres <andrespach4@gmail.com>
-Co-authored-by: psantollo <psantollo@ucol.mx>
\ No newline at end of file
+Co-authored-by: psantollo <psantollo@ucol.mx>
```
Comment (vzqzac) [inline team-commit.txt:18]: I think this file change should not be part of the commit

**Review by vzqzac — APPROVED** [review]
Great cleanup! Just please take the team-commit.txt change out of the commit and we're good to go
